certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/certificateOutput.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.certifikit.cli
+
+import app.cash.certifikit.BasicConstraints
+import app.cash.certifikit.Certificate
+import app.cash.certifikit.ObjectIdentifiers
+import java.security.cert.X509Certificate
+import java.time.Instant.ofEpochMilli
+import javax.net.ssl.X509TrustManager
+import okhttp3.internal.platform.Platform
+import okio.ByteString
+import okio.ByteString.Companion.toByteString
+import picocli.CommandLine.Help.Ansi
+
+fun X509Certificate.sha256Hash(): ByteString =
+  publicKey.encoded.toByteString()
+      .sha256()
+
+fun Certificate.prettyPrintCertificate(
+  trustManager: X509TrustManager = Platform.get()
+      .platformTrustManager()
+): String {
+  val sha256 = this.sha256
+
+  return buildString {
+    val trustedRoot = sha256 != null && trustManager.acceptedIssuers.find {
```

Comment:
I’ve done some reading recently and chainbuilding is really difficult to do correctly. In particular there can be cross-signed things and verifications that reject certain edges.

This is probably fine for the CLI, but not enough for actual trust.

Maybe more words would help (“signed by locally-trusted root”)

Reply:
I added because I wanted to build a tool that would help with basic understanding and manipulating of certificates.  Without learning both openssl and JDK command syntax.  I'll spin off a task to do this properly including loading from another target JDK/keystore.